sale_layout_category_hide_detail/views/sale_views.xml

```diff
@@ -0,0 +1,71 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2019 Tecnativa - Ernesto Tejeda
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <record id="view_order_form" model="ir.ui.view">
+        <field name="name">sale.order.hide.detail.form</field>
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='order_line']/form//field[@name='name']" position="after">
+                <group>
+                    <field name="show_details"
+                           attrs="{'invisible': [('display_type', '!=', 'line_section')]}"/>
+                    <field name="show_subtotal"
+                           attrs="{'invisible': ['|', ('display_type', '!=', 'line_section'), ('show_details', '!=', True)]}"/>
+                </group>
+            </xpath>
+            <xpath expr="//field[@name='order_line']/tree" position="inside">
+                <field name="show_details"
+                       widget="boolean_fa_icon"
+                       options="{
+                            'show_in_line_section': true,
+                            'fa_icons': {
+                                'icon_true': 'fa-eye',
+                                'icon_false': 'fa-eye-slash',
+                            },
+                            'terminology': {
+                                'hover_true': 'Switch to: details hidden',
+                                'hover_false': 'Switch to: details shown',
+                            }
+                       }"
+                       attrs="{'invisible': [('display_type', '!=', 'line_section')]}"/>
+                <field name="show_subtotal"
+                       widget="boolean_fa_icon"
+                       options="{
+                            'show_in_line_section': true,
+                            'fa_icons': {
+                                'icon_true': 'fa-plus-circle',
+                                'icon_false': 'fa-minus-circle',
+                            },
+                            'terminology': {
+                                'hover_true': 'Switch to: subtotal hidden',
+                                'hover_false': 'Switch to: subtotal shown',
```

Comment:
You didn't consider translations here. You can get inspired on [how the `boolean_button` widget handles this](https://github.com/odoo/odoo/blob/674ee035ca954a57341cbaa951d2f6f31b5f8fe2/addons/web/static/src/js/fields/basic_fields.js#L2112-L2133).

Reply:
boolean_toggle is a FieldBoolean with the checkbox made up with scss, but this new widget is not rendered as a checkbox and its behavior is not the same as the FieldBoolean. If I inherited from a FieldBoolean I would have to rewrite almost all FieldBoolean methods to disable its behavior, and then the inheritance would no longer make sense. It is the same as what happens with FieldBooleanButton, so that inherits from AbstractField instead of FieldBoolean